lib/karafka/callbacks.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module Karafka
+  # Additional callbacks that are used to trigger some things in given places during the
+  # system lifecycle
+  # @note Those callbacks aren't the same as consumer calbacks as they are not related to the
+  #   lifecycle of particular messages fetches but rather to the internal flow process.
+  #   They cannot be defined on a consumer callback level, because for some of those,
+  #   there aren't consumers in the memory yet and/or they aren't per consumer thread
```

Comment:
Those callbacks aren't the same as consumer callbacks as they are not related to the lifecycle of particular messages fetches but rather to the internal flow process. They cannot be defined on a consumer callback level because for some of those, there aren't consumers in the memory yet and/or they aren't per consumer thread